recipes/install.rb

```diff
@@ -26,10 +26,13 @@
   version = install['version']
   edition = install['edition']
   url = source_download_url(version, edition)
+  # allow to override the install dir on a per-edition basis
+  install_directory = node['visualstudio'][version][edition]['install_dir'] || node['visualstudio'][version]['install_dir'] # rubocop:disable Metrics/LineLength
```

Comment:
This implies there could be both a `community` and `buildtools` editions installed ... how does the machine handle that when they're both 2017, won't there be overlap?

Reply:
Unsure, but I know I can't install both microsoft build tools and the agents for visual studio in the same directory. Teamcity identifies the separate paths correctly. So it ends up being `../2017/BuildTools` and `../2017/TestAgent`. I verified that our builds appear to still work with both installed.

Reply:
Shouldn't then the recipe just raise an exception if _both_ are defined?